### ISSUE TITLE
p tags are stripped in WYSIWYG fields

### ISSUE THREAD

**Issue by GoogleCodeExporter** [viktorthang/wordpress-custom-content-type-manager#292]
```
1. clean install of wordpress 3.3.1
2. clean install of CCTM 0.9.5
3. create a custom content type with CCTM
4. add a wysiwyg field
5. add a copy of the example template to the twentyeleven theme
6. add a page of the created custom content type to the site
7. copy and paste a piece of text with paragraphs in both the first and the 
second wysiwyg field
8. view the page, and check the source code, the second field strips out all p 
tags


*SYSTEM INFO* 
------------------------ 
Plugin Version: 0.9.5.5-pl
WordPress Version: 3.3.1
PHP Version: 5.3.8
MySQL Version: 5.0.51a-24+lenny5-log
Server OS: Linux
------------------------ 
Other Active plugins: 


```

Original issue reported on code.google.com by `3tree...@gmail.com` on 17 Jan 2012 at 11:55

**Comment by GoogleCodeExporter** [comment]
```
I'll look into this, but just a heads up: WP implementation of WYSIWYG and the 
TinyMCE editor is *terrible*.  Version 3.3 finally saw some sensible 
development, but the implementation is still completely myopic: the WP devs 
apparently never considered the possibility that the editor would be used for 
anything other than the main content field.  Previous versions of the CCTM had 
limitations in the WYSIWYG fields because of this, and I don't expect a fully 
clean solution until WP cleans up their source code, unfortunately.
```

Original comment by `ever...@fireproofsocks.com` on 17 Jan 2012 at 4:12
- Changed state: **InReview**

**Comment by GoogleCodeExporter** [comment]
```
Clarifying, paragraph tags are stripped from ALL WYSIWYG fields, no matter 
whether the CCTM is active or not.  This appears to be the "standard" WP 
behavior.  To verify this, go to a fresh install of WP without CCTM installed, 
and try pasting some HTML into the content editor using the HTML editor, e.g.

<p>Hello.
<br/>
Again.</p>
<span>Testing...</span>

Then toggle to the visual editor.  POOF.  Toggling nukes your paragraph tags. 

So really what we're talking about is the behavior that should take over if the 
p tags have been nuked.  For the main content field, newlines are converted to 
p or br tags via the wpautop() function: 
http://codex.wordpress.org/Function_Reference/wpautop

To work around this, you can update your templates to something like this:

<strong>My WYSIWYG:</strong> <?php print 
wpautop(get_custom_field('mywysiwyg:do_shortcode')); ?>

See how that works?  If you think this should ALWAYS happen to all custom 
WYSIWYG fields, then maybe we can change this to a feature request, but I tend 
to think that the user should consciously do this in their templates instead of 
dictating it upstream.
```

Original comment by `ever...@fireproofsocks.com` on 18 Jan 2012 at 8:40
- Changed title: **p tags are stripped in WYSIWYG fields**
- Changed state: **Invalid**

**Comment by GoogleCodeExporter** [comment]
```
Have been testing with the wpautop() function, but no succes. Checked if the 
remove_filter was active, but is was not. Tested this in a fresh WP3.3.1 
installation with CCTM 0.9.5
```

Original comment by `3tree...@gmail.com` on 20 Jan 2012 at 9:06

**Comment by GoogleCodeExporter** [comment]
```
What's your code look like?  Remember that wpautop() doesn't print its output, 
it only returns it.  Make sure you have debug mode active in your wp-config.php 
file too so you can identify any errors.
```

Original comment by `ever...@fireproofsocks.com` on 20 Jan 2012 at 4:45

**Comment by GoogleCodeExporter** [comment]
```
Ok, how I implemented this was inside the do_shortcode filter.  You can 
override the behavior if needed, but what we're looking at now is that if your 
WYSIWYG fields use the do_shortcode output filter.  THis is on the dev branch 
now, will be included in 0.9.5.6.
```

Original comment by `ever...@fireproofsocks.com` on 23 Jan 2012 at 7:42
- Changed state: **Fixed**
